### PR TITLE
feat(autoware_v2x_msgs): add virtual gate messages

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Add virtual gate message. See [README](https://github.com/isamu-takagi/autoware_msgs/blob/feat/add-v2x-messages/autoware_v2x_msgs/README.md) for details.

## Related links

https://github.com/autowarefoundation/autoware.universe/issues/2703

## Tests performed

Not applicable.

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by dimethylether261 — APPROVED** [review]

**Review by mitsudome-r — APPROVED** [review]

**Review by yukkysaito — APPROVED** [review]
test approval

**Comment by evshary** [comment]
It's a good design. Will the PR be merged in the roadmap?
Also, if I understand correctly, the router in the v2x component will be implemented in the future, right?

**Comment by isamu-takagi** [comment]
>It's a good design. Will the PR be merged in the roadmap?

Thank you. Since it has been discussed several times, I think it would be a good idea to merge and try it out.
@mitsudome-r @yukkysaito What do you think? Would it be better to separate the package so that it can be seen that it is a prototype message because it has not been sufficiently tested?

>Also, if I understand correctly, the router in the v2x component will be implemented in the future, right?

Yes, the v2x component does not exist yet, so we need to create a new one.

File: autoware_v2x_msgs/package.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>autoware_v2x_msgs</name>
+  <version>1.0.0</version>
+  <description>Autoware common messages package.</description>
+  <maintainer email="isamu.takagi@tier4.jp">Takagi, Isamu</maintainer>
```
Comment (yukkysaito) [inline autoware_v2x_msgs/package.xml:7]: Could you add @yukkysaito  and @mitsudome-r  to the maintainers?

**Comment by mitsudome-r** [comment]
Once the maintainer information is updated, the PR looks good to me.

**Review by mitsudome-r — APPROVED** [review]

File: autoware_v2x_msgs/package.xml (hunk 1 above)
Comment (isamu-takagi) [reply autoware_v2x_msgs/package.xml:7]: @yukkysaito Add maintainters https://github.com/autowarefoundation/autoware_msgs/pull/77/commits/35755d7bb3b6ed2da71c32d0779e23f4e41ea4b0.

File: autoware_v2x_msgs/msg/VirtualGateAreaCommand.msg [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# constants for command
+uint16 ACQUIRE = 1
+uint16 RELEASE = 2
+
+# variables for lock
+uint16   command
+uint16   sequence  # Used to check correspondence with commands.
```
Comment (mitsudome-r) [inline autoware_v2x_msgs/msg/VirtualGateAreaCommand.msg:7]:
```suggestion
uint16   sequence_id  # Used to check correspondence with status messages.
```

File: autoware_v2x_msgs/msg/VirtualGateAreaCommand.msg [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+string   id        # Target area ID.
```
Comment (mitsudome-r) [inline autoware_v2x_msgs/msg/VirtualGateAreaCommand.msg:8]:
```suggestion
string   area_id        # Target area ID.
```

File: autoware_v2x_msgs/msg/VirtualGateAreaCommand.msg [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+string[] gates     # Entry and exit gate ID.
```
Comment (mitsudome-r) [inline autoware_v2x_msgs/msg/VirtualGateAreaCommand.msg:9]:
```suggestion
string[] gate_ids     # Entry and exit gate ID.
```

File: autoware_v2x_msgs/msg/VirtualGateAreaStatus.msg [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+uint16 RESERVED = 1
+uint16 ACQUIRED = 2
+uint16 RELEASED = 3
+
+# variables for lock
+builtin_interfaces/Time stamp
+uint16   status
+uint16   sequence  # Used to check correspondence with commands.
+string   id        # Target area ID.
+string[] gates     # Entry and exit gate ID.
```
Comment (mitsudome-r) [inline autoware_v2x_msgs/msg/VirtualGateAreaStatus.msg:11]:
```suggestion
uint16   sequence_id  # Used to check correspondence with commands.
string   area_id        # Target area ID.
string[] gate_ids     # Entry and exit gate ID.
```

File: autoware_v2x_msgs/msg/VirtualGateCommand.msg [hunk 6]
```diff
@@ -0,0 +1,3 @@
+builtin_interfaces/Time stamp
+autoware_v2x_msgs/VirtualGateAreaCommand[] areas
+autoware_v2x_msgs/VirtualGateVehicleInfo vehicle
```
Comment (yukkysaito) [inline autoware_v2x_msgs/msg/VirtualGateCommand.msg:3]: memo
will remove.
Because of the separate location information

**Comment by mitsudome-r** [comment]
Maybe we can change the interface name to something like:
* Lock Road Interface
* Acquire Road Interface
* Request to Pass Interface

File: autoware_v2x_msgs/msg/VirtualGateCommand.msg (hunk 6 above)
Comment (isamu-takagi) [reply autoware_v2x_msgs/msg/VirtualGateCommand.msg]: Removed https://github.com/autowarefoundation/autoware_msgs/pull/77/commits/3e7f9d553c5d777bd84eb523b696482d2c02254a.

**Comment by isamu-takagi** [comment]
> Maybe we can change the interface name to something like:

@mitsudome-r @yukkysaito Similar messages are "Signal Request Message" and "Signal Status Message" called in ITS standards. So what about the following names?
- RoadLockRequest.msg / RoadLockStatus.msg
- AcquireRoadRequest.msg / AcquireRoadStatus.msg
- RightOfWayRequest.msg / RightOfWayStatus.msg
- PassPermitRequest.msg / PassPermitStatus.msg

**Comment by mitsudome-r** [comment]
I don't mind with virtual gate messages for now.

**Pull request merged**